### DISCUSSION TITLE
Ability to set the chunk size of a selectinload

### DISCUSSION THREAD

**Discussion by ollz272** [sqlalchemy/sqlalchemy#11448] [Usage Questions]
Hi,

We want to be able to customise the chunk size of selectin load. This is set on a code level at 500: https://github.com/sqlalchemy/sqlalchemy/blob/6f704cac01a0b10e7c4531e20f8edd47f25c150f/lib/sqlalchemy/orm/strategies.py#L2839

We'd love to be able to change this, either as a project default, or on a case by case basis. Maybe something like:
```python
selectinload(Table.relationship, chunk_size=1000)
selectinload(Table.relationship).options(chunk_size=1000)
```

If there are any other ways i'd love to hear.

**Comment by zzzeek** [comment]
additional params can be added to options like selectinload() but what about relationship(lazy='selectin') ?

**Reply by CaselIT** [reply]
something like `relationship(lazy=('selectin', {'chunk_size': 1000}))` ?
Maybe making a function that generates these options similar to backref? So something like
`relationship(lazy=loader_options('selectin', {'chunk_size': 1000}))`

**Reply by zzzeek** [reply]
we've never done it like that before.  the current way relationship takes arguemtns like this is as paramters directly such as https://docs.sqlalchemy.org/en/20/orm/relationship_api.html#sqlalchemy.orm.relationship.params.join_depth  https://docs.sqlalchemy.org/en/20/orm/relationship_api.html#sqlalchemy.orm.relationship.params.innerjoin https://docs.sqlalchemy.org/en/20/orm/relationship_api.html#sqlalchemy.orm.relationship.params.omit_join etc. so this would just be another parameter straight on relationship for now, until we want to do some different system for these

**Reply by CaselIT** [reply]
Indeed the precedent is that. Likely that's the best way. Will create an issue for this

**Reply by CaselIT** [reply]
Tracked here https://github.com/sqlalchemy/sqlalchemy/issues/11450
